packages/typespec-vscode/src/npm-utils.ts

```diff
@@ -0,0 +1,150 @@
+import { readFile } from "fs/promises";
+import path from "path";
+import semver from "semver";
+import logger from "./log/logger.js";
+import { ExecOutput, loadModule, spawnExecution, spawnExecutionEvents } from "./utils.js";
+
+export enum InstallationAction {
```

Comment:
usually we prefer to use union to enum.  like:   InstallationAction = "install" | "upgrade" | "skip" | "cancel". same for below

Reply:
I want to use the enum to define the supported action, user don't need to pass in the value. Union defined is to define an type or string literal, it may not well control/guide.